reimplemenation with many new features

mwinter69 opened the pull request:
This reimplementation brings in the following features:
- python2 and python3 support
- Python2/3 installation to configure location of python, with auto detect feature
- inline python scripts
- use python script from file sytem
- python options and script arguments

It is backward compatible with the existing python plugin. Old calls of the plugin are translated by default to python 2 inline scripts without any options or arguments. With a marker file in JENKINS_HOME a conversion to python 3 is possible.

This reimplementation solves the following issues:
[JENKINS-7551]
[JENKINS-12483]
[JENKINS-29007]

mwinter69 committed:
Merge pull request #1 from jenkinsci/master

merge master to private branch

mwinter69 committed:
reimplementation of python plugin

Features:
- Support for python2 and python3
- allow to give a path to a file or enter directly a script
- define options for python
- define arguments for the script

mwinter69 committed:
formatting of code

mwinter69 committed:
adjust python version print so it works with <2.7

mwinter69 committed:
resolve env vars

jhasse commented:
> It is backward compatible with the existing python plugin. Old calls of the plugin are translated to python 2 inline scripts without any options or arguments.

Unfortunately this will break my build scripts: I'm using the existing Python plugin with Python 3.

mwinter69 committed:
marker file for conversion to new plugin

if t marker file is present, existing scripts will be converted to python 3
otherwise to python 2

mwinter69 commented:
Solved this with a marker file. Put the file python3CompytibilityMarker in JENKINS_HOME and they are converted to python 3

jhasse commented:
I'd rather not have to create a file in my JENKINS_HOME. Also some people won't know about this when updating the plugin.

I think the right approach would be to change `wantedVersion` from an `int` to a `String` and set it to `""` for old calls of the plugin. This way it will use the `python` executable in the `PATH`, just like it does now.
